tests/vm_watch_test.go

```diff
@@ -0,0 +1,334 @@
+		Expect(vmStatus.name).To(Equal("NAME"))
+		Expect(vmStatus.age).To(Equal("AGE"))
+		Expect(vmStatus.running).To(Equal("RUNNING"))
+		Expect(vmStatus.volume).To(Equal("VOLUME"))
+
+		// Read first status of the vm
+		vmStatus = readVMStatus(stdout, vmStatus, readTimeout)
+		Expect(vmStatus.name).To(Equal(vm.Name))
+		Expect(vmStatus.running).To(Equal("false"))
+
+		By("Starting the VM")
+		StartVirtualMachine(vm)
+
+		vmStatus = readVMStatus(stdout, vmStatus, readTimeout)
+		Expect(vmStatus.running).To(Equal("true"))
+
+		By("Restarting the VM")
+		err = virtCli.VirtualMachine(vm.ObjectMeta.Namespace).Restart(vm.ObjectMeta.Name)
+		Expect(err).To(BeNil())
+
+		vmStatus = readVMStatus(stdout, nil, readTimeout)
+		Expect(vmStatus.running).To(Equal("true"))
+
+		By("Stopping the VM")
+		vm = StopVirtualMachine(vm)
+
+		vmStatus = readVMStatus(stdout, vmStatus, readTimeout)
+		Expect(vmStatus.running).To(Equal("false"))
+
+		By("Deleting the VM")
+		err = virtCli.VirtualMachine(vm.ObjectMeta.Namespace).Delete(vm.ObjectMeta.Name, &v1.DeleteOptions{})
+		Expect(err).To(BeNil())
+	})
+
+	It("Should update vmi 'PHASE' when using 'kubectl get vmi -w'", func() {
+		By("Creating a random VMI spec")
+		vm := NewRandomVMWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskCirros))
+		Expect(vm).To(Not(BeNil()))
+
+		By("Setting up the kubectl command")
+		cmd, stdout, stderr :=
+			createCommandWithNSAndRedirect(vm.ObjectMeta.Namespace, GetK8sCmdClient(), "get", "vmi", "-w")
+		Expect(cmd).To(Not(BeNil()))
+
+		err = cmd.Start()
+		Expect(err).To(BeNil())
+
+		defer cmd.Process.Kill()
+
+		go failOnError(stderr)
+
+		time.Sleep(processWaitTime)
+
+		By("Applying vmi to the cluster")
+		vm, err = virtCli.VirtualMachine(vm.ObjectMeta.Namespace).Create(vm)
+		Expect(err).To(BeNil())
+
+		// Start a VMI
+		vm = StartVirtualMachine(vm)
+
+		// Read the column titles
+		vmiStatus := readVMIStatus(stdout, nil, readTimeout)
+		Expect(vmiStatus.name).To(Equal("NAME"))
+		Expect(vmiStatus.age).To(Equal("AGE"))
+		Expect(vmiStatus.phase).To(Equal("PHASE"))
+		Expect(vmiStatus.ip).To(Equal("IP"))
+		Expect(vmiStatus.node).To(Equal("NODENAME"))
+
+		vmiStatus = readVMIStatus(stdout, vmiStatus, readTimeout)
+		Expect(vmiStatus.phase).To(Equal(""))
+
+		vmiStatus = readVMIStatus(stdout, vmiStatus, readTimeout)
+		Expect(vmiStatus.phase).To(Equal("Pending"))
+
+		vmiStatus = readVMIStatus(stdout, vmiStatus, readTimeout)
+		Expect(vmiStatus.phase).To(Equal("Scheduling"))
+
+		vmiStatus = readVMIStatus(stdout, vmiStatus, readTimeout)
+		Expect(vmiStatus.phase).To(Equal("Scheduled"))
+
+		vmiStatus = readVMIStatus(stdout, vmiStatus, readTimeout)
+		Expect(vmiStatus.phase).To(Equal("Running"))
+
+		// Restart the VMI
+		err = virtCli.VirtualMachine(vm.ObjectMeta.Namespace).Restart(vm.ObjectMeta.Name)
+		Expect(err).To(BeNil())
+
+		vmiStatus = readVMIStatus(stdout, vmiStatus, readTimeout)
+		Expect(vmiStatus.phase).To(Equal("Failed"))
```

Comment:
Do we really enter a FAILED state when we restart? is it a known bug or expected behavior?

Reply:
The test plan specifies 'Failed' as an expected result, I will look into it.

Reply:
I checked and `Failed` is a valid status since the VM was forcefully restarted

Reply:
@omeryahud You can add description to the assertion if it is needed. Next to the "HaveOccured()", the second argument of ToNot can be a string that describes the assertion reason.

Reply:
@enp0s3 Done, thanks
@danielBelenky We have discussed this offline